var/spack/repos/builtin/packages/elfutils/package.py

```diff
@@ -73,11 +73,13 @@ class Elfutils(AutotoolsPackage, SourcewarePackage):
     depends_on("zlib", type="link")
     depends_on("gettext", when="+nls")
     depends_on("m4", type="build")
+    depends_on("pkgconfig@0.9.0:", type=("build", "link"))
 
     # debuginfod has extra dependencies
-    # NB: Waiting on an elfutils patch before we can use libmicrohttpd@0.9.71
-    depends_on("libmicrohttpd@0.9.33:0.9.70", type="link", when="+debuginfod")
-    depends_on("libarchive@3.1.2:", type="link", when="+debuginfod")
+    # NB: Waiting on an elfutils patch before we can use libmicrohttpd@0.9.51
+    depends_on("libmicrohttpd@0.9.33:0.9.50", type="link", when="+debuginfod")
+    # libarchive with iconv doesn't configure: https://github.com/spack/spack/issues/36710
+    depends_on("libarchive@3.1.2: ~iconv", type="link", when="+debuginfod")
```

Comment:
https://github.com/spack/spack/issues/36710#issuecomment-1502117314:
> I can confirm that the problem initially presented itself with libarchive@3.6.2. Using the previous version allows elfutils to build. 

 maybe we should make this version dependent then?

Reply:
Sure.

Reply:
Made `libarchive@3.6.2 +iconv` a `conflict`. Building against `libarchive@3.5.2+iconv` works for me as well, and the concretizer is smart enough to figure out `libarchive@3.6.2 ~iconv` by default.